src/main/webapp/app/entities/result/result.component.ts

```diff
@@ -128,65 +127,49 @@ export class ResultComponent implements OnInit, OnChanges {
-        } else {
-            // Evaluate the template status for quiz and programming exercise.
-            this.templateStatus = this.evaluateTemplateStatusForQuizProgrammingExercises();
-        }
-    }
-
-    /**
-     * Evaluates the template status modeling, text and file upload exercises
-     *
-     * @return {ResultTemplateStatus}
-     */
-    private evaluateTemplateStatusForModelingTextFileUploadExercises() {
-        if (!this.participation) {
-            // this should NOT happen, but we provide a fallback case
+        // Fallback if participation is not set
+        if (!this.participation || !getExercise(this.participation)) {
             if (!this.result) {
                 return ResultTemplateStatus.NO_RESULT;
             } else {
                 return ResultTemplateStatus.HAS_RESULT;
             }
         }
-        const submissionInDueTime =
-            !getExercise(this.participation).dueDate ||
-            (this.participation.submissions != null &&
-                this.participation.submissions.length > 0 &&
-                isSubmissionInDueTime(this.participation.submissions[0], getExercise(this.participation)));
-        const assessmentDueDate = this.dateAsMoment(getExercise(this.participation).assessmentDueDate!);
 
-        // Submission is in due time of exercise and has a result with score.
-        if (submissionInDueTime && initializedResultWithScore(this.result)) {
-            // Prevent that the result is shown before assessment due date
-            return !assessmentDueDate || assessmentDueDate.isBefore() ? ResultTemplateStatus.HAS_RESULT : ResultTemplateStatus.NO_RESULT;
-        } else if (submissionInDueTime && !initializedResultWithScore(this.result)) {
-            // Submission is in due time of exercise and doesn't have a result with score.
-            return ResultTemplateStatus.SUBMITTED;
-        } else if (initializedResultWithScore(this.result) && (!assessmentDueDate || assessmentDueDate.isBefore())) {
-            // Submission is not in due time of exercise, has a result with score and there is no assessmentDueDate for the exercise or it lies in the past.
-            return ResultTemplateStatus.LATE;
-        } else {
-            // Submission is not in due time of exercise and tThere is actually no feedback for the submission or the feedback should not be displayed yet.
-            return ResultTemplateStatus.LATE_NO_FEEDBACK;
+        // Evaluate status for modeling, text and file-upload exercises
+        if (isModelingOrTextOrFileUpload(this.participation)) {
+            // Based on its submission we test if the participation is in due time of the given exercise.
+            const inDueTime = isParticipationInDueTime(this.participation, getExercise(this.participation));
+            const assessmentDueDate = this.dateAsMoment(getExercise(this.participation).assessmentDueDate!);
```

Comment:
what happens if assessment due date is null?

Reply:
I use `assessmentDueDate` only for conditional checks in the following lines:

- `return !assessmentDueDate || assessmentDueDate.isBefore() ? ResultTemplateStatus.HAS_RESULT : ResultTemplateStatus.NO_RESULT;`
- `} else if (initializedResultWithScore(this.result) && (!assessmentDueDate || assessmentDueDate.isBefore())) {`

As far as I know `!assessmentDueDate` evaluates to `true` if it is `null` or `undefined`. As logical operators are executed *short-circuit* in JavaScript, the right side of `||` will never be checked if the left side is `true` (-> `null` or `undefined`).

Reply:
@sascha11110 I think you still can remove `! ` here to not get somebody confused to think that `assessmentDueDate` can't be null

Reply:
@maxr96 Maybe I just don’t  get it :D But if I remove the `!` in the following expression
`return !assessmentDueDate || assessmentDueDate.isBefore() ? ResultTemplateStatus.HAS_RESULT : ResultTemplateStatus.NO_RESULT;` then the second part of the condition `assessmentDueDate.isBefore()` would possibly fail as there is no implicit `null` check via the logical operator.

Reply:
@sascha11110 No I meant here: `const assessmentDueDate = this.dateAsMoment(getExercise(this.participation).assessmentDueDate!);`

Reply:
Thanks for pointing this out! I checked that `dateAsMoment()` is able to handle `null` values and removed the exclamation mark.